### PR TITLE
Findevent

### PR DESCRIPTION
**Pull request by arnaudlvq**
In this PULL REQUEST, i made the following changes :

- MapContainer was moved from a Data folder to a UI folder.
- The Map is displayed in the findEvent Screen
- Search Bar from Material 3 has been added to the findEvent Screen as well (style not finished yet)

### CODE REVIEW

**Review by juaniac — CHANGES_REQUESTED** [review]
Overall great job, here are a few adjustments you should make so that this works with what the view model and database provide:

-Instead of searching for keywords, make it search for the title of the event as the functionality of searching for keywords (something in the title or description) is something too time consuming to implement right away.

-Instead of selecting the exact date of an event, you should add the functionality of having a valid range of dates. The user should input their availability (for example from Friday at 17h00 to Monday at 2h00) using date/time pickers so that they can have a wider range of an events they can find. 

-You should add a location search + a search radius in km, so that the user can search for events near them or near a place that they will be going to.

If you need more detail go check out the implementation of the find an event view model.

Also please fix the merge errors before asking for a pull request so that they can be verified before merging with the main.

**Comment by arnaudlvq** [comment]
The code now includes
- a slider allowing the user to select the event search zone around a location
- a button to a DatePicker to select the start date of the desired event (new process discussed live with Juaniac)
- the map now automatically zooms around markers

File: app/src/main/java/com/monkeyteam/chimpagne/model/location/LocationHelper.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import org.json.JSONArray
+
+class LocationHelper {
+
+  private val _markers = MutableStateFlow<List<Location>>(emptyList())
+  val markers: StateFlow<List<Location>> = _markers.asStateFlow()
+
+  suspend fun convertNameToLocation(name: String, onResult: (Location) -> Unit) {
+    val client = OkHttpClient()
+    val urlSearchNominatim = "https://nominatim.openstreetmap.org/search?q=$name&format=json"
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/model/location/LocationHelper.kt:21]: Should be declared as a constant

File: app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+
+  var tagInput by remember { mutableStateOf("") }
+  val selectedTags = remember { mutableStateListOf<String>() }
+
+  var searchRadius by remember { mutableStateOf(1f) }
+
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            title = { Text("Find an event") },
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt:126]: Use strings defined in resources for text, this allows for translations

File: app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+                  Modifier.fillMaxSize().padding(16.dp).verticalScroll(rememberScrollState()),
+              horizontalAlignment = Alignment.Start) {
+                FindEventLegend("Event Location", Icons.Rounded.Search, "Search")
+
+                Spacer(Modifier.height(16.dp))
+
+                LocationFinder("", {}, emptyList(), {}, Modifier.fillMaxWidth())
+                Spacer(Modifier.height(16.dp))
+                IconTextButton(
+                    text = "Locate me",
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt:157]: Use strings defined in resources for text, this allows for translations

File: app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+                Text(text = "Search Radius: ${searchRadius.toInt()} km")
+                Slider(
+                    value = searchRadius,
+                    onValueChange = { searchRadius = it },
+                    valueRange = 1f..30f,
+                    modifier = Modifier.fillMaxWidth())
+
+                Spacer(Modifier.height(40.dp))
+
+                FindEventLegend("Add Tags to your search", Icons.Rounded.Tag, "Tags")
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt:172]: Use strings defined in resources for text, this allows for translations

File: app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+                    keyboardActions =
+                        KeyboardActions(
+                            onDone = {
+                              if (tagInput.trim().isNotBlank()) {
+                                selectedTags.add(0, tagInput.trim().lowercase(Locale.getDefault()))
+                                tagInput = ""
+                              }
+                            }),
+                    maxLines = 1,
+                    placeholder = { Text("Enter a tag...") },
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt:188]: Use strings defined in resources for text, this allows for translations

File: app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+
+                Spacer(Modifier.height(16.dp))
+
+                Row(modifier = Modifier.fillMaxWidth().horizontalScroll(rememberScrollState())) {
+                  selectedTags.forEach { tag -> TagChip(tag = tag) { selectedTags.remove(tag) } }
+                }
+
+                Spacer(Modifier.height(40.dp))
+
+                FindEventLegend("Date of the event", Icons.Rounded.CalendarToday, "Select date")
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt:199]: Use strings defined in resources for text, this allows for translations

File: app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt [hunk 7]
```diff
@@ -0,0 +1,370 @@
...
+
+  BottomSheetScaffold(
+      sheetContent = {
+        Box(Modifier.fillMaxWidth().height(128.dp), contentAlignment = Alignment.Center) {
+          Text("Swipe up to expand sheet")
+        }
+        Column(
+            Modifier.fillMaxWidth().padding(64.dp),
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              Text("Sheet content")
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt:317]: Use strings defined in resources for text, this allows for translations

File: app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt [hunk 8]
```diff
@@ -0,0 +1,370 @@
...
+
+  val addMarker = { location: Location ->
+    coroutineScope.launch { locationHelper.addMarker(location) }
+  }
+
+  val removeMarker = { location: Location ->
+    coroutineScope.launch { locationHelper.removeMarker(location) }
+  }
+
+  addMarker(Location("Balelec", 46.519144, 6.566804))
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt:303]: This should be done in a test

**Review by Athanaze — COMMENTED** [review]
I annotated some of the hardcoded string that should be changed, there are more in other files too

**Comment by arnaudlvq** [comment]
Now, when a marker is clicked, the map is lifted and the marker is centered, while the bottom sheet is expanding.

File: app/src/main/java/com/monkeyteam/chimpagne/model/location/LocationHelper.kt (hunk 1 above)
Comment (arnaudlvq) [reply app/src/main/java/com/monkeyteam/chimpagne/model/location/LocationHelper.kt:21]: domain is now saved in a constant

File: app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt (hunk 4 above)
Comment (arnaudlvq) [reply app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt]: Done

File: app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt (hunk 3 above)
Comment (arnaudlvq) [reply app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt]: Done

File: app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt (hunk 6 above)
Comment (arnaudlvq) [reply app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt]: Done

File: app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt (hunk 7 above)
Comment (arnaudlvq) [reply app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt]: Done

File: app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt (hunk 8 above)
Comment (arnaudlvq) [reply app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt]: Thanks!

**Pull request merged**